File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    answer = []
    for i in range(0, len(arr)):
        if flag[i]:
            for j in range(0, arr[i]*2):
                answer.append(arr[i])
        else:
            for j in range(0, arr[i]):
                answer.pop()
    return answer

File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        int[] answer = {};
        List<Integer> arrX = new ArrayList<Integer>();
        for(int i=0;i<arr.length;i++){
            if(flag[i]){
                for(int j=0;j<arr[i]*2;j++){
                    arrX.add(arr[i]);    
                }
            } else{
                for(int j=0;j<arr[i];j++){
                    arrX.remove(arrX.size()-1);
                }
            }
        }
        return arrX.stream().mapToInt(i->i).toArray();
    }
}